vulcanai2/models/basenetwork.py

```diff
@@ -289,12 +285,10 @@ def _get_criterion(criterion_spec):
         criterion_spec = pdash.omit(criterion_spec, "name")
         return criterion_class(**criterion_spec)
 
-    def _init_trainer(self):
+    def _init_trainer(self, train_loader):
         self.optim = self._init_optimizer(self._optim_spec)
         self.criterion = self._get_criterion(self._criter_spec)
-
-        self.valid_interv = 2*len(self.train_loader)
-        self.epoch = 0
+        self.valid_interv = 2*len(train_loader)
```

Comment:
we can remove the self.valid_interv since it is now just passed to .fit. would simplify the function signature to remove it as well